extensions/positron-python/python_files/posit/positron/matplotlib_backend.py

```diff
@@ -163,7 +163,12 @@ def render(self, size: PlotSize | None, pixel_ratio: float, format_: str) -> byt
 
         # This must be set before setting the size and can't be passed via print_figure else the
         # resulting size won't match the request size.
-        self.figure.set_layout_engine("tight")
+        try:
+            # plotnine plots use their own layout engine, do not override
+            if self.figure._layout_engine.__module__ == "plotnine._mpl.layout_engine":
```

Comment:
the retina problems were stemming from us overriding `plotnine`'s layout manager! updated so we don't override it

Reply:
Great catch! Can we use the public method `self.figure.get_layout_engine()`? Might also need to check that it isn't `None`. And it might be safer to do something like `.startswith('plotnine')` in case they move it around

Reply:
what do we think of _always_ leaving the layout engine alone if it is not None? I understand the sizing won't be exact, but it might be a simple escape hatch to respect user's settings.

Reply:
Sounds like a better approach! We'd just need to double-check that it doesn't break any existing behavior. Might be covered by e2e tests.